cleaning

flexsurfer opened the pull request:
cleaning (removed unused files in status-im, mostly for chat context), introduce react-native.red-black-tree and move messages list events
![image](https://user-images.githubusercontent.com/11790366/212914694-8abce43b-0b3d-4c1d-9d22-bfa38a465c0d.png)

flexsurfer committed:
cleaning, introduce react-native.red-black-tree and move messages list events

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 5f468125 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14808/1/) | 2023-01-17 13:45:24 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14808/1/consoleText) |

flexsurfer committed:
lint and test

ilmotta commented:
Awesome stuff here, thank you! I'll review this PR later today @flexsurfer.

status-im-auto commented:
## 65% of end-end tests have passed
```
Total executed tests: 23
Failed tests: 8
Passed tests: 15
```
```
IDs of failed tests: 702732,702775,702745,702851,702731,702808,702838,702843 
```
<h3>Failed tests (8)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14808.apk&PR_ID=14808&TR_CASE_IDS=702732,702775,702745,702851,702731,702808,702838,702843">Rerun failed tests</a></li><br/><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075943">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: SignInButton </div><div>## Signed in successfully!</div></blockquote></p>

```
medium/test_deeplink_chat_share_profile.py:394: in test_deep_link_open_user_profile
    chat.wait_for_element_starts_with_text(ens_user['username'])
../views/base_view.py:467: in wait_for_element_starts_with_text
    return element.wait_for_element(wait_time)
../views/base_element.py:116: in wait_for_element
    raise TimeoutException(
 Device `1`: `Button` by` xpath`: `//*[starts-with(@text,'Legal Vibrant Indianabat')]` is not found on the screen after wait_for_element 
```

<b>[[Profile is often not opened in e2e builds for some reason. Needs to be investigated.]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c778723db444bcbbdc6167bcb9e622c?auth=34ec177c2633dd83c942a68850eeeed5#98">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/7c778723db444bcbbdc6167bcb9e622c/0029screenshot.png?auth=34ec177c2633dd83c942a68850eeeed5">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075941">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `MemberPhoto` by `xpath`: `//*[starts-with(@text,'profile_photo')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='user-avatar']` </div><div>Device 2: Image differs from template to 41.807502297794116 percents </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1465: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Image of user in 1-1 chat is too different from template! 
```

<b>[[On profile picture failed due to #14718]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#167">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8/0072screenshot.png?auth=a7d53da4c783c77e28a0b5538849f6a3">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#108">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db/0066screenshot.png?auth=e2621a6067a1a3cfe0146f912b1d8983">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075938">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><p><blockquote><div>Device 1: Wait for text element `Button` to be equal to `2` </div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='pins-count']/android.widget.TextView` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1345: in test_1_1_chat_pin_messages
    chat.pinned_messages_count.wait_for_element_text("2",
../views/base_element.py:151: in wait_for_element_text
    text_element = self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='pins-count']/android.widget.TextView` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#145">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8/0072screenshot.png?auth=a7d53da4c783c77e28a0b5538849f6a3">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#101">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db/0066screenshot.png?auth=e2621a6067a1a3cfe0146f912b1d8983">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075959">test_activity_center_mentions_in_community_jump_to</a>, id: 702851 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[contains(@text, 'Corny Standard Seal')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` </div><div>Device 1: Find `Button` by `accessibility id`: `close-activity-center` </div></blockquote></p>

```
medium/test_activity_center.py:247: in test_activity_center_mentions_in_community_jump_to
    self.home_1.handle_contact_request(username=self.default_username_2)
../views/home_view.py:363: in handle_contact_request
    self.close_activity_centre.click()
../views/base_element.py:90: in click
    self.find_element().click()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `close-activity-center` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/747318c180364550ba83b108e905bbdc?auth=7d0b9f878695f7e28888cf87a1ec26d4#86">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/747318c180364550ba83b108e905bbdc/0021screenshot.png?auth=7d0b9f878695f7e28888cf87a1ec26d4">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/21e123d20081486da7f6fb78d49db31e?auth=297e5b48a80d7fb8ade0d5c8aab500c8#68">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/21e123d20081486da7f6fb78d49db31e/0022screenshot.png?auth=297e5b48a80d7fb8ade0d5c8aab500c8">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075939">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div># STEP: Check pinned messages count and content</div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='pins-count']/android.widget.TextView` </div></blockquote></p>

```
critical/chats/test_group_chat.py:370: in test_group_chat_pin_messages
    count = group_chat.pinned_messages_count.text
../views/base_element.py:208: in text
    return self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='pins-count']/android.widget.TextView` is not found on the screen ; also Unexpected Alert is shown: 'Rendered more hooks than during the previous render.'
```

<b>[[blocked by #14637]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9e4ae996ef84441598b88a2fadd62b80?auth=76c7134189e65e05e4aaa260dab41709#67">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/9e4ae996ef84441598b88a2fadd62b80/0038screenshot.png?auth=76c7134189e65e05e4aaa260dab41709">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3b25780991cb4a48860f85a82cf24e47?auth=f101fa2d30868f5f23cc6d17b52e53f5#60">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/3b25780991cb4a48860f85a82cf24e47/0017screenshot.png?auth=f101fa2d30868f5f23cc6d17b52e53f5">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075950">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 1: Looking for a message by text: hello </div><div>Device 1: Find `TimeStampText` by `xpath`: `//*[starts-with(@text,'hello')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-timestamp']` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:625: in test_community_message_send_check_timestamps_sender_username
    (sent_time_variants.join(','), timestamp))
 'list' object has no attribute 'join'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#133">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7/0065screenshot.png?auth=5d920d28113aceb2ace6042310c16879">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#86">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc/0035screenshot.png?auth=58b24e72eaf2735097a27411e56abd53">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075955">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `SendMessageButton` by `accessibility id`: `send-message-button` </div><div>Device 1: Tap on found: SendMessageButton </div></blockquote></p>

```
critical/test_public_chat_browsing.py:642: in test_community_message_edit
    self.errors.verify_no_errors()
base_test_case.py:188: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message is not edited
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#139">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7/0065screenshot.png?auth=5d920d28113aceb2ace6042310c16879">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#89">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc/0035screenshot.png?auth=58b24e72eaf2735097a27411e56abd53">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075948">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `ChatButton` by `accessibility id`: `start-group-chat-button` </div></blockquote></p>

```
critical/chats/test_group_chat.py:248: in test_group_chat_offline_pn
    self.homes[0].create_group_chat(user_names_to_add=[self.usernames[1], self.usernames[2]],
../views/home_view.py:394: in create_group_chat
    chat_view = self.new_group_chat_button.click()
../views/base_element.py:90: in click
    self.find_element().click()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: ChatButton by accessibility id: `start-group-chat-button` is not found on the screen 
```

<b>[[mysterious issue when PNs are not fetched from offline,can not reproduce on real devices; needs investigation]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/588fd4618ac44777a528d9bd31db678c?auth=ddb42af6e3f4b853869c1a10464bb3aa#127">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/588fd4618ac44777a528d9bd31db678c/0036screenshot.png?auth=ddb42af6e3f4b853869c1a10464bb3aa">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e5d3fdda370f4c8c8b777a73820a2104?auth=0d25258f65c233fc2f6c84625fd7e432#79">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/e5d3fdda370f4c8c8b777a73820a2104/0021screenshot.png?auth=0d25258f65c233fc2f6c84625fd7e432">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5ac5aba3b5b24a8bba6d9302fadb06f3?auth=fc0200a2f1566e4036726e5b4aea36c2#68">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/5ac5aba3b5b24a8bba6d9302fadb06f3/0018screenshot.png?auth=fc0200a2f1566e4036726e5b4aea36c2">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (15)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075947">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/588fd4618ac44777a528d9bd31db678c?auth=ddb42af6e3f4b853869c1a10464bb3aa#113">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e5d3fdda370f4c8c8b777a73820a2104?auth=0d25258f65c233fc2f6c84625fd7e432#63">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5ac5aba3b5b24a8bba6d9302fadb06f3?auth=fc0200a2f1566e4036726e5b4aea36c2#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075944">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c778723db444bcbbdc6167bcb9e622c?auth=34ec177c2633dd83c942a68850eeeed5#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075942">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c778723db444bcbbdc6167bcb9e622c?auth=34ec177c2633dd83c942a68850eeeed5#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075957">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1b24468e4dc24a9ea215615e39051d59?auth=54f599fe50192b76719edd96270c5b29#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075958">test_activity_center_decline_contact_request_no_pn</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/747318c180364550ba83b108e905bbdc?auth=7d0b9f878695f7e28888cf87a1ec26d4#50">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/21e123d20081486da7f6fb78d49db31e?auth=297e5b48a80d7fb8ade0d5c8aab500c8#38">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075952">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#176">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#134">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075956">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#244">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#163">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075953">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#191">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#147">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075951">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#154">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#105">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075954">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a93df1245705444a8e6e72f941c5c9c7?auth=5d920d28113aceb2ace6042310c16879#224">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/917e8f7839cc481abeeba70737d141cc?auth=58b24e72eaf2735097a27411e56abd53#155">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075940">test_1_1_chat_text_message_edit_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#291">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#199">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075937">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#83">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#69">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075945">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#105">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075949">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#251">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#173">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3075946">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0581da8a65124d6090371a5782bdeae8?auth=a7d53da4c783c77e28a0b5538849f6a3#341">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/56ddce7fee324b4986b7bc5e80ad98db?auth=e2621a6067a1a3cfe0146f912b1d8983#247">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

ilmotta commented on src/mocks/js_dependencies.cljs:
```diff
@@ -1,8 +1,7 @@
   (:require [status-im.fleet.default-fleet :refer (default-fleets)])
-  (:require [status-im.utils.test :as utils.test])
-  (:require [status-im.chat.default-chats :refer (default-chats)]))
+  (:require [status-im.utils.test :as utils.test]))
```
Would be nice to move this `utils.test` namespace somewhere else. This file has side-effects even to load it, and we can see from the only 2 usages in the codebase, that it should be required only once.

A better place for this file is in `status-im2/setup`, where we have defined in the guidelines to put some files related to raw plumbing. Another alternative would be to move it to `src/test_helpers/` directory because the integration tests use the `init!` function to define poll for signals.

flexsurfer replied:
i agree, but its kinda not related to this PR, i just removed one require line here

flexsurfer replied:
i was trying to keep this PR only for chat context

ilmotta replied:
Agree, this is more of tangential point.

flexsurfer commented:
> Awesome stuff here, thank you! I'll review this PR later today @flexsurfer.

there is not much to review, i just removed a lot of old code, and moved one namespace, just moved rn library in separate ns

ilmotta commented on src/react_native/red_black_tree.cljs:
```diff
@@ -0,0 +1,51 @@
+(ns react-native.red-black-tree
+  (:refer-clojure :exclude [remove update find])
+  (:require ["functional-red-black-tree" :as rb-tree]))
```
In the spirit of our guidelines, perhaps we should use the alias as `red-black-tree`, there's no need to shorten the alias.

ilmotta commented on src/react_native/red_black_tree.cljs:
```diff
@@ -0,0 +1,51 @@
+  [^js tree item]
+  (.find tree item))
+
+(defn insert
```
Really nice to see this refactor, but IMO this namespace has nothing to do with React Native. Putting it inside `react_native/` implies there's some relationship, but there's none. How about we move it to `src/utils/red_black_tree.cljs`?

flexsurfer replied:
really good point thanks

ilmotta commented on src/status_im/chat/models/message_test.cljs:
```diff
@@ -1,148 +0,0 @@
-(ns status-im.chat.models.message-test
```
Why are we removing this test file? Shouldn't we adapt it?

flexsurfer replied:
mhm good question

flexsurfer replied:
removed it by mistake, thanks. reverted

ilmotta replied:
Thanks Andrey

ilmotta commented on src/status_im2/contexts/chat/messages/list/events.cljs:
```diff
@@ -1,12 +1,12 @@
-  (:require ["functional-red-black-tree" :as rb-tree]
-            [status-im.constants :as constants]
+(ns status-im2.contexts.chat.messages.list.events
+  (:require [react-native.red-black-tree :as rb-tree]
```
I think we should stick to our convention of preferring explicit namespace aliases, so in this case: `red-black-tree` instead of `rb-tree`.

ilmotta reviewed (approved):
@flexsurfer, although it's pretty much impossible to review a PR of this size, it looks good to me :rocket: 

I have left a couple comments that I would like to see addressed, but I'm okay if you need to merge this PR faster to avoid conflicts and work on the changes in a separate PR.

flexsurfer committed:
fixe
